File: tests/cases/go/basics.expected.go
package main

import (
	"fmt"
	"strings"
)

type Point struct {
	X, Y int
	Name string `json:"name"`
}

type Shape interface {
	Area() float64
}

func (p *Point) Move(dx, dy int) error {
	if dx < 0 {
		return fmt.Errorf("neg %d", dx)
	} else if dy < 0 {
		return nil
	}
	for i := 0; i < 10; i++ {
		p.X += i
	}
	for _, v := range items {
		total += v
	}
	switch v := x.(type) {
	case int:
		fmt.Println("int")
	default:
		fmt.Println("other")
	}
	go func() {
		ch <- 1
	}()
	defer func() { recover() }()
	m := map[string]int{
		"a": 1,
		"b": 2,
	}
	select {
	case msg := <-ch:
		fmt.Println(msg)
	case <-done:
		return nil
	}
	if err := do(); err != nil {
		return err
	}
	s := strings.Join([]string{
		"a",
		"b",
	}, ",")
	return nil
}

File: tests/cases/rust/basics.expected.rs
use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct Point<'a> {
    pub x: i32,
    name: &'a str,
}

impl<'a> Point<'a> {
    pub fn new(x: i32, name: &'a str) -> Self {
        Self { x, name }
    }

    fn describe(&self) -> String {
        match self.x {
            0 => "zero".to_string(),
            n if n < 0 => {
                let s = format!("neg {}", n);
                s
            }
            _ => format!("{} at {}", self.name, self.x),
        }
    }
}

pub trait Shape {
    fn area(&self) -> f64;
}

enum Msg {
    Quit,
    Move { x: i32, y: i32 },
    Write(String),
}

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut map: HashMap<String, Vec<u8>> = HashMap::new();
    for (k, v) in map.iter() {
        println!("{k}: {v:?}");
    }
    if let Some(x) = opt {
        x.run()?;
    } else if cond {
        return Ok(());
    }
    while let Some(item) = it.next() {
        total += item;
    }
    let v: Vec<_> = (0..10).filter(|x| x % 2 == 0).map(|x| {
        x * 2
    }).collect();
    'outer: loop {
        break 'outer;
    }
    let raw = r#"a "quoted" {brace}"#;
    let c = 'x';
    let closure = |a: i32, b: i32| -> i32 { a + b };
    Ok(())
}

mod tests {
    #[test]
    fn it_works() {
        assert_eq!(2 + 2, 4);
    }
}
